libraries/AP_RangeFinder/AP_RangeFinder_LeddarOne.cpp

```diff
@@ -48,36 +48,67 @@ bool AP_RangeFinder_LeddarOne::detect(RangeFinder &_ranger, uint8_t instance, AP
 // read - return last value measured by sensor
 bool AP_RangeFinder_LeddarOne::get_reading(uint16_t &reading_cm)
 {
+    uint8_t number_detections;
+    LeddarOne_Status leddarone_status;
+
     if (uart == nullptr) {
         return false;
     }
 
-    // send a request message for Modbus function 4
-    if (send_request() != LEDDARONE_OK) {
-        // TODO: handle LEDDARONE_ERR_SERIAL_PORT
-        return false;
-    }
+    switch (modbus_status) {
 
-    uint32_t start_ms = AP_HAL::millis();
-    while (!uart->available()) {
-        // wait up to 200ms
-        if (AP_HAL::millis() - start_ms > 200) {
-            return false;
-        }
-    }
+        case LEDDARONE_MODBUS_PRE_SEND_REQUEST:
+            // clear buffer and buffer_len
+            memset(read_buffer, 0, sizeof(read_buffer));
```

Comment:
if you have a lose connection then your state machine will resend and continually clear this buffer. There should be an additional state of "init" which is set on a complete packet but skipped if we get no data. To make it faster, that init state will set to the next state and then fall through to that next state of sending but a repeat send on timeout would skip the init